src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private final double firstLeg;
+    private final double secondLeg;
+    private final ColorSupplier color;
```

Comment:
The `color` should not be of type `ColorSupplier`. Instead, it should be of type `Color`. `ColorSupplier` is used to generate a random color, but the color itself should be stored as an enum type `Color`.